File: 50 Coding Challenge Part I/Python/Challenge12.py
"""
Challenge 12a: Create a function that receives an array of numbers and returns an array containing only the positive numbers.
 * Requirement: Use a “for” loop
"""
def positive_numbers(numbers):
    positive = []
    for i in numbers:
        if i >= 0:
            positive.append(i)
    return positive


array = [-5, 10, -3, 12, -9, 5, 90, 0, 1]
print(positive_numbers(array))


"""
@project 12b: Create a function that receives an array of numbers and returns an array containing only the negative numbers.
 * Requirement: Use a “for” loop
"""
def negative_numbers(numbers):
    negative = []
    for i in numbers:
        if i < 0:
            negative.append(i)
    return negative


array = [-5, 10, -3, 12, -9, 5, 90, 0, 1]
print(negative_numbers(array))


"""
@project 12c: Create a function that receives an array of numbers and returns an array containing only the positive numbers.
 * Requirement: Use .filter() array method
"""
def filtered_numbers(numbers):
    positive = filter(lambda i: i >= 0, numbers)
    return positive


array = [-5, 10, -3, 12, -9, 5, 90, 0, 1]
print(list(filtered_numbers(array)))

File: 50 Coding Challenge Part I/C/challenge12.c
// Coding challenge 12: Create a function that recieves an array of numbers and 
// returns an array containing only positive numbers

#include <stdio.h>

int * getPositive(int arr[], int len);

int main(int argc, char *argv[]) {

	int arr[] = {-5, 10, -3, 12, -9, 5, 90, 0, 1};
	int len = sizeof(arr)/sizeof(arr[0]);
	int * arrPointer = getPositive(arr, len);

	for(int i = 0; i < len; ++i) {
		printf("*(p + %d): %d\n", i, arrPointer[i]);
	}

	return 0;
}

int * getPositive(int arr[], int len) {
	static int positive[10];
	int index = 0;

	for(int i = 0; i < len; ++i) {
		if(arr[i] >= 0) {
			positive[index] = arr[i];
			++index;
		}
	}

	return positive;
}
